site/cds_rdm/tasks.py

```diff
@@ -140,3 +147,224 @@ def sync_users():
     except Exception as e:
         db.session.rollback()
         current_app.logger.exception(e)
+
+
+@shared_task()
+def sync_local_accounts_to_names(since, user_id=None):
+    """Syncs local accounts to names vocabulary.
+
+    If the name is marked as "unlisted" - meaning it is deprecated - we still update it with the new values, if needed.
+    """
+    prop_values = ["group", "department", "maibox", "section"]
+    service = current_service_registry.get("names")
+
+    def _check_if_update_needed(user, name, is_orcid=False):
+        """Check if the name needs to be updated."""
+
+        updated = False
+        updated_name = {**name}
+
+        if not is_orcid:
+            if user.user_profile.get("given_name") != name.get("given_name"):
+                updated_name["given_name"] = user.user_profile.get("given_name")
+                updated = True
+
+            if user.user_profile.get("family_name") != name.get("family_name"):
+                updated_name["family_name"] = user.user_profile.get("family_name")
+                updated = True
+
+        user_affiliation = user.user_profile.get("affiliations", "")
+        name_affiliations = [aff["name"] for aff in name.get("affiliations", [])]
+        if user_affiliation and user_affiliation not in name_affiliations:
+            if "affiliations" not in updated_name:
+                updated_name["affiliations"] = []
+            updated_name["affiliations"].append({"name": user_affiliation})
+            updated = True
+
+        user_orcid = user.user_profile.get("orcid")
+        name_orcids = [
+            identifier["identifier"]
+            for identifier in name.get("identifiers", [])
+            if identifier["scheme"] == "orcid"
+        ]
+        if user_orcid and user_orcid not in name_orcids:
+            if "identifiers" not in updated_name:
+                updated_name["identifiers"] = []
+            updated_name["identifiers"].append(
+                {"scheme": "orcid", "identifier": user_orcid}
+            )
+            updated = True
+
+        for prop in prop_values:
+            user_prop = user.user_profile.get(prop)
+            name_prop = name.get("props", {}).get(prop)
+            if user_prop and user_prop != name_prop:
+                if "props" not in updated_name:
+                    updated_name["props"] = {}
+                updated_name["props"][prop] = user_prop
+                updated = True
+
+        # add the email as props
+        if user.email and user.email != name.get("props", {}).get("email"):
```

Comment:
how is email then consumed?

Reply:
The email is displayed in the UI 
![image](https://github.com/user-attachments/assets/3d8c68ad-9165-4c1d-a73a-521053d47727)